pkg/azuremachine/validate_update.go

```diff
@@ -51,6 +51,11 @@ func (a *UpdateValidator) Validate(ctx context.Context, request *v1beta1.Admissi
 		return false, microerror.Maskf(errors.ParsingFailedError, "unable to parse AzureMachine CR: %v", err)
 	}
 
+	err := checkSSHKeyIsEmpty(ctx, AzureMachineNewCR)
```

Comment:
I know it's not from this PR but `AzureMachineNewCR` should start with lowercase.

Reply:
renamed